### DISCUSSION TITLE
Using Conventional Commits

### DISCUSSION THREAD

**Discussion by aymanbagabas** [go-git/go-git#933] [General]
I've seen a lot of projects use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) to construct commit messages. Go-git has its own format that is _very_ similar to [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/). I'm suggesting that we make the transition since it will help with constructing and automating coming up with a changelog.

Proposed changes:

Use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) to write commit messages with concrete types that clearly describe the intent for the commit i.e. `fix`, `feat`, `docs`, `refactor`, etc.

Current format:

```
<package>: <subpackage>, <what changed>. [Fixes #<issue-number>]

// message
```

Suggested format:

```
<type>(<package>): [optional subpackage]: <description>

// message

Fixes: <issue-url> / #<issue-number>
```

This uses [git trailer format](https://git-scm.com/docs/git-interpret-trailers) to include additional links, issue numbers, and signatures.

cc/ @mcuadros @pjbgf

**Comment by pjbgf** [comment]
I agree that by adding the additional type of the change, it would make it a lot easier for users to understand the changes without having to dive into each one of the PR/issues.
